File: backend/src/models/Deal.ts
import {
    Table,
    Column,
    CreatedAt,
    UpdatedAt,
    Model,
    PrimaryKey,
    AutoIncrement,
    ForeignKey,
    BelongsTo,
    DataType,
    Default
} from "sequelize-typescript";
import Contact from "./Contact";
import Ticket from "./Ticket";
import Pipeline from "./Pipeline";
import PipelineStage from "./PipelineStage";
import Tenant from "./Tenant";

@Table
class Deal extends Model<Deal> {
    @PrimaryKey
    @AutoIncrement
    @Column
    id: number;

    @Column
    title: string;

    @Default(0.00)
    @Column(DataType.DECIMAL(10, 2))
    value: number;

    @Default(1)
    @Column
    priority: number;

    @ForeignKey(() => Contact)
    @Column
    contactId: number;

    @BelongsTo(() => Contact)
    contact: Contact;

    @ForeignKey(() => Ticket)
    @Column
    ticketId: number;

    @BelongsTo(() => Ticket)
    ticket: Ticket;

    @ForeignKey(() => Pipeline)
    @Column
    pipelineId: number;

    @BelongsTo(() => Pipeline)
    pipeline: Pipeline;

    @ForeignKey(() => PipelineStage)
    @Column
    stageId: number;

    @BelongsTo(() => PipelineStage)
    stage: PipelineStage;

    @ForeignKey(() => Tenant)
    @Column(DataType.UUID)
    tenantId: number | string;

    @BelongsTo(() => Tenant)
    tenant: Tenant;

    @CreatedAt
    createdAt: Date;

    @UpdatedAt
    updatedAt: Date;
}

export default Deal;

File: bussines/internal/models/deal.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Deal struct {
	ID        int       `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"not null" json:"name"`
	Value     float64   `gorm:"type:decimal(10,2);default:0" json:"value"`
	Status    string    `gorm:"default:'open'" json:"status"`
	StageID   int       `gorm:"column:stageId" json:"stageId"`
	ContactID int       `gorm:"column:contactId" json:"contactId"`
	TicketID  *int      `gorm:"column:ticketId" json:"ticketId"`
	TenantID  uuid.UUID `gorm:"column:tenantId;type:uuid" json:"tenantId"`
	CreatedAt time.Time `gorm:"column:createdAt" json:"createdAt"`
	UpdatedAt time.Time `gorm:"column:updatedAt" json:"updatedAt"`

	// Relations
	Contact Contact       `gorm:"foreignKey:ContactID" json:"contact,omitempty"`
	Stage   PipelineStage `gorm:"foreignKey:StageID" json:"stage,omitempty"`
}

func (Deal) TableName() string {
	return "Deals"
}
